File: library/src/main/java/org/incendo/interfaces/interfaces/ChestInterfaceBuilder.java
package org.incendo.interfaces.interfaces;

import net.kyori.adventure.text.Component;
import org.incendo.interfaces.pane.ChestPane;

public final class ChestInterfaceBuilder extends AbstractInterfaceBuilder<ChestPane, ChestInterface> {

    private int rows = 0;
    private Component initialTitle = null;

    public int rows() {
        return this.rows;
    }

    public ChestInterfaceBuilder rows(final int rows) {
        this.rows = rows;
        return this;
    }

    public Component initialTitle() {
        return this.initialTitle;
    }

    public ChestInterfaceBuilder initialTitle(final Component initialTitle) {
        this.initialTitle = initialTitle;
        return this;
    }

    @Override
    public ChestInterface build() {
        return new ChestInterface(
            this.rows,
            this.initialTitle,
            this.closeHandlers(),
            this.transforms(),
            this.clickPreprocessors()
        );
    }
}

File: next/src/main/java/org/incendo/interfaces/next/interfaces/ChestInterfaceBuilder.kt
package org.incendo.interfaces.next.interfaces

import net.kyori.adventure.text.Component
import org.incendo.interfaces.next.pane.ChestPane

public class ChestInterfaceBuilder :
    AbstractInterfaceBuilder<ChestPane, ChestInterface>() {

    public var rows: Int = 0
    public var initialTitle: Component? = null

    public override fun build(): ChestInterface = ChestInterface(
        rows,
        initialTitle,
        transforms,
        clickPreprocessors
    )
}
